src/environmentd/src/bin/environmentd/main.rs

```diff
@@ -630,7 +634,21 @@ fn run(mut args: Args) -> Result<(), anyhow::Error> {
     {
         cors::Any.into()
     } else if !args.cors_allowed_origin.is_empty() {
-        AllowOrigin::list(args.cors_allowed_origin)
+        let allowed = args.cors_allowed_origin.clone();
+        AllowOrigin::predicate(move |origin: &HeaderValue, _request_parts: _| {
+            for val in &allowed {
+                // For each specified allow cors origin, check if it starts with
+                // a *. and allow anything from that glob. Otherwise check for
+                // an exact match.
+                if (val.as_bytes().starts_with(b"*.")
```

Comment:
I think this is somewhat weird, in that IIUC origins usually start with "http://" or "https://". Will be maybe weird to have this discrepancy, I think?

```
environmentd --allow-origin=https://cloud.materialize.com
environmentd --allow-origin=*.materialize.vercelapp.com
```

Reply:
Yes, that's how it would look. My first iteration required that if `*.blah` was present, it was also the only member, so we wouldn't have that. During a huddle Robin and I thought that we might actually want to have both, because today we use staging and localhost, which that would have broken.

Reply:
It's really too bad that [`wildmatch`](https://github.com/becheran/wildmatch) doesn't work on byte strings, because I think it's exactly what you want here otherwise.

Reply:
Oh, sorry, what I'm trying to get at is that I think in an ideal world you'd be able to say `https://*.materialize.vercelapp.com` (note the https:// prefix).